pkg/proxy_test.go

```diff
@@ -0,0 +1,93 @@
+package pkg
```

Comment:
Awesome, thanks for providing the tests!
What about testing the TTL expiration? You could simply repeat GetMetricById call with a sleep in the test. 
Similar test: https://github.com/app-sre/aws-resource-exporter/blob/master/pkg/cache_test.go#L47

Reply:
Great point. I'll incorporate that shortly